src/pinns_pde_solve.jl

```diff
@@ -1206,13 +1360,113 @@ function SciMLBase.discretize(pde_system::PDESystem, discretization::PhysicsInfo
         (pde_loss_functions, bc_loss_functions)
     end
 
-    pde_loss_function = θ -> sum(map(l->l(θ) ,pde_loss_functions))
-    bcs_loss_function = θ -> sum(map(l->l(θ) ,bc_loss_functions))
-    loss_function = θ -> pde_loss_function(θ) + bcs_loss_function(θ)
+    # setup for all adaptive losses
+    num_pde_losses = length(pde_loss_functions)
+    num_bc_losses = length(bc_loss_functions)
+    # assume one single additional loss function if there is one. this means that the user needs to lump all their functions into a single one,
+    num_additional_loss = additional_loss isa Nothing ? 0 : 1 
+
+    adaloss_T = eltype(adaloss.pde_loss_weights)
+    logger = discretization.logger
+    log_frequency = discretization.log_options.log_frequency
+    iteration = discretization.iteration
+    self_increment = discretization.self_increment
+
+    # this will error if the user has provided a number of initial weights that is more than 1 and doesn't match the number of loss functions
+    adaloss.pde_loss_weights = ones(adaloss_T, num_pde_losses) .* adaloss.pde_loss_weights
+    adaloss.bc_loss_weights = ones(adaloss_T, num_bc_losses) .* adaloss.bc_loss_weights
+    adaloss.additional_loss_weights = ones(adaloss_T, num_additional_loss) .* adaloss.additional_loss_weights
+
+    function logvector(v, name)
+        if logger isa TBLogger
+            for j in 1:length(v)
+                log_value(logger, "$(name)/$(j)", v[j], step=iteration[1])
+            end
+        end
+    end
+
+    function logscalar(s, name)
+        if logger isa TBLogger
+            log_value(logger, "$(name)", s, step=iteration[1])
+        end
+    end
+
+    # this is the function that gets called to do the adaptive reweighting, a function specific to the 
+    # type of adaptive reweighting being performed. 
+    # TODO: I'd love to pull this out and then dispatch on it via the AbstractAdaptiveLoss, so that users can implement their own
+    #       currently this is kind of tricky since the different methods need different types of information, and the loss functions
+    #       are generated internal to the code
```

Comment:
Open an issue and let's do that in a further iteration.

Reply:
https://github.com/SciML/NeuralPDE.jl/issues/489#issue-1154686102